packages/datadog-plugin-jest/test/circus.spec.js

```diff
@@ -236,6 +251,52 @@ describe('Plugin', function () {
         )
       })
 
+      it('can report code coverage', function (done) {
+        let contentTypeHeader,
+          coveragePayload,
+          contentDisposition,
+          eventContentDisposition,
+          eventPayload
+
+        const scope = nock(`http://127.0.0.1:${agent.server.address().port}`)
+          .post('/api/v2/citestcov')
+          .reply(202, function () {
+            contentTypeHeader = this.req.headers['content-type']
+            contentDisposition = this.req.requestBodyBuffers[1].toString()
+            eventContentDisposition = this.req.requestBodyBuffers[6].toString()
+            eventPayload = this.req.requestBodyBuffers[8].toString()
+            coveragePayload = msgpack.decode(this.req.requestBodyBuffers[3])
+          })
+
+        const options = {
+          ...jestCommonOptions,
+          testRegex: 'jest-coverage.js',
+          coverage: true
+        }
+
+        jestExecutable.runCLI(
+          options,
+          options.projects
+        ).then(() => {
+          // it takes a bit for the payload to be flushed
+          setTimeout(() => {
+            expect(scope.isDone()).to.be.true
+            expect(contentTypeHeader).to.contain('multipart/form-data')
+            expect(coveragePayload.version).to.equal(1)
+            expect(coveragePayload.files.map(file => file.filename))
+              .to.include('packages/datadog-plugin-jest/test/sum-coverage-test.js')
+            expect(contentDisposition).to.contain(
+              'Content-Disposition: form-data; name="coverage1"; filename="coverage1.msgpack"'
+            )
+            expect(eventContentDisposition).to.contain(
+              'Content-Disposition: form-data; name="event"; filename="event.json"'
+            )
+            expect(eventPayload).to.equal('{}')
+            done()
+          }, 2000)
```

Comment:
Rather than a timeout, couldn't you just put all these checks inside the nock reply handler?